### DISCUSSION TITLE
Issue with yarn install when packages are linked to git submodules

### DISCUSSION THREAD

**Discussion by Kankarollo** [renovatebot/renovate#31457] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

38.66.0

### Please tell us more about your question or problem

In my company we've stumbled upon special case in which renovate is not working correctly. 
Some frontend teams are linking through path aliasing packages to git submodules directories. 
```
...
"compilerOptions": {
  "baseUrl": ".",
  "rootDir": ".",
  "paths": {
    "@abc/shared/*": [
    "./project/submodules/*"
    ],
...
``` 
In submodules there are git submodules.

Sadly that creates error because renovate does not recognize this and don't install these modules before executing yarn install command. Because of that in all of these projects we are receiving error message: 
```
warning Resolution field "husky@7.0.4" is incompatible with requested version "husky@^1.3.1"
error Couldn't find package "@abc/shared@*" required by "@abc/project-submodule@0.1.0" on the "npm" registry.
```
Tried to use git-submodules manager in renovate but sadly it only creates PRs with bump of submodules and don't resolve the issue. Best solution would be to have this preUpgradeTask but I see that's still in planning phase https://github.com/renovatebot/renovate/issues/8804.

Is there some other workaround available to handle this issue?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Sounds like you need cloneSubmodules, which is one of the first hits when you search Renovate docs for "submodules": <img width="635" alt="image" src="https://github.com/user-attachments/assets/cff67dcf-5916-4d3e-a623-8395506a8eac">

**Comment by Kankarollo** [comment]
Sorry for creating stupid issue, indeed there is this option that probably solves my issue. I've focused on using git-submodules and didn't think about investigating into documentation more.  I'm testing it right now.
I'm receiving error:
```
"err": {
         "task": {
           "commands": ["checkout", "-f", "--recurse-submodules", "dev", "--"],
           "format": "utf-8",
           "parser": "[function]"
         },
"message": "fatal: not a git repository: ../../../.git/modules/project-name/submodules/submodule-name\nfatal: could not reset submodule index\n"
```
But I'm not sure if it's still renovate issue or repository configuration. Anyway will create seperate discussion if that's renovate issue. 

Thanks for quick answer.
